### PR TITLE
Feature: Display Itemized Country of Origin Information

### PR DESCRIPTION
**Pull request by pasindu-clouda**
#### What does this PR do? \*

- add itemized country of origin details

workspace: [https://quickorderdev--sbdsefuat.myvtex.com](https://quickorderdev--sbdsefuat.myvtex.com/quickorder?__bindingAddress=catalog.sef.ao.sbd-nonprod.com/en-US/catalog-global)

### CODE REVIEW

File: react/TextAreaBlock.tsx [hunk 1]
```diff
@@ -81,6 +82,7 @@ const TextAreaBlock: FunctionComponent<
   const [isModalOpen, setIsModelOpen] = useState<boolean>(false)
   const [base64Image, setBase64Image] = useState('')
   const [excelDownloading, setExcelDownloading] = useState<boolean>(false)
+  // const [countryOfOriginList, setCountryOfOriginList] = useState<any>([])
```
Comment (neeshsilva) [inline react/TextAreaBlock.tsx]: Remove unwanted commented code lines.

File: react/components/ReviewBlock.tsx [hunk 2]
```diff
@@ -472,6 +471,8 @@ const ReviewBlock: FunctionComponent<WrappedComponentProps & any> = ({
         const sellers = item.sku ? mappedRefId[item.sku]?.sellers : '1'
         const itm = getItemFromQuery(item)
 
+        console.log('itm', itm)
```
Comment (neeshsilva) [inline react/components/ReviewBlock.tsx]: Remove the console logs.

**Review by neeshsilva — CHANGES_REQUESTED** [review]
Update the change log.

File: node/resolvers/search/index.ts [hunk 3]
```diff
@@ -498,4 +502,18 @@ export const queries = {
...
+    } = ctx
+
+    const res = await masterdata.searchDocumentsWithPaginationInfo({
+      dataEntity: COUNTRY_OF_ORIGIN_ACRONYM,
+      schema: COUNTRY_OF_ORIGIN_SCHEMA,
+      fields: COUNTRY_OF_ORIGIN_FIELDS,
+      pagination: { pageSize: 1000, page: 1 },
+    })
+    return res.data
+  },
```
Comment (Nadeera-Gayashan) [inline node/resolvers/search/index.ts:582]: can't we narrow down the search using 'where' condition? also reduce the page size at least to 10.

File: react/TextAreaBlock.tsx [hunk 4]
```diff
@@ -523,6 +525,23 @@ const TextAreaBlock: FunctionComponent<
...
+  //     const { data } = await refetch()
+  //     const countryOfOrigin = data?.getCountryOfOrigin ?? []
+  //     setCountryOfOriginList(countryOfOrigin)
+  //   }
+
+  //   if (countryOfOriginList.length === 0) {
+  //     refetchCountryOfOrigin
+  //   }
+  // }, [])
+
```
Comment (Nadeera-Gayashan) [inline react/TextAreaBlock.tsx:546]: remove unwanted comments

File: react/utils/const.ts [hunk 5]
```diff
@@ -129,3 +129,1186 @@ export const SAMPLE_TABLE = [
     width: 15,
   },
 ]
+
+export const COO_DATA = [
```
Comment (Nadeera-Gayashan) [inline react/utils/const.ts]: why these data?

**Review by Nadeera-Gayashan — CHANGES_REQUESTED** [review]

File: react/utils/const.ts (hunk 5 above)
Comment (pasindu-clouda) [reply react/utils/const.ts]: oh we can remove those

File: node/resolvers/search/index.ts (hunk 3 above)
Comment (pasindu-clouda) [reply node/resolvers/search/index.ts:582]: ahh okay

File: react/TextAreaBlock.tsx [hunk 6]
```diff
@@ -560,7 +581,10 @@ const TextAreaBlock: FunctionComponent<
             ? `${item.JDE_Weight} ${item.JDE_Weight_UOM}/${item.JDE_Weight_Per_UOM}`
             : ' ',
           tariffCode: item.JDE_HTS_Code,
-          origin: item.JDE_Country_of_Origin,
+          origin:
+            countryOfOriginList.find(
```
Comment (Nadeera-Gayashan) [inline react/TextAreaBlock.tsx]: null check?

File: react/TextAreaBlock.tsx [hunk 7]
```diff
@@ -585,7 +609,10 @@ const TextAreaBlock: FunctionComponent<
...
```
Comment (Nadeera-Gayashan) [inline react/TextAreaBlock.tsx]: null check?

File: react/components/ReviewBlock.tsx [hunk 8]
```diff
@@ -644,6 +652,18 @@ const ReviewBlock: FunctionComponent<WrappedComponentProps & any> = ({
           const itemError = getLineError(rowData)
           const itemAvailability = getLineItemStatus(rowData)
 
+          const tooltipTitle =
+            countryOfOriginList.find(
```
Comment (Nadeera-Gayashan) [inline react/components/ReviewBlock.tsx]: null check?

**Review by Nadeera-Gayashan — CHANGES_REQUESTED** [review]

File: react/components/ReviewBlock.tsx [hunk 9]
```diff
@@ -644,6 +652,18 @@ const ReviewBlock: FunctionComponent<WrappedComponentProps & any> = ({
...
+              (coo) => coo.udc === rowData.JDE_Country_of_Origin
+            )?.text ?? rowData.JDE_Country_of_Origin
+
+          const coo = truncateText(
+            countryOfOriginList.find(
```
Comment (Nadeera-Gayashan) [inline react/components/ReviewBlock.tsx]: countryOfOriginList?.find

File: react/TextAreaBlock.tsx [hunk 10]
```diff
@@ -523,18 +525,37 @@ const TextAreaBlock: FunctionComponent<
...
+        tableData
+          .filter((item: any) => item?.JDE_Country_of_Origin)
+          .map((item: any) => item.JDE_Country_of_Origin)
+      )
+    }
+  }, [tableData])
+
   const downloadExcelFile = async () => {
     setExcelDownloading(true)
     const data = tableData.flatMap((item: any) => {
```
Comment (Nadeera-Gayashan) [inline react/TextAreaBlock.tsx]: Here

**Review by Nadeera-Gayashan — CHANGES_REQUESTED** [review]

**Review by Nadeera-Gayashan — APPROVED** [review]

**Pull request merged**